### ISSUE TITLE
rust-analyzer 0.3.1631-standalone can't not expand proc_macro 

### ISSUE THREAD

**Issue by qinchuan-dev** [rust-lang/rust-analyzer#15509]
cargo 1.71.1 (7f1d04c00 2023-07-29)
rustc 1.71.1 (eb26296b5 2023-08-03)
CARGO_HOME=~/.cargo
RUSTUP_HOME=~/.rustup

https://github.com/dtolnay/syn/blob/master/examples/trace-var/example/src/main.rs

rust-analyzer can not expand the 'trace_var' macro correctly:
source code:
```rust
use trace_var::trace_var;

fn main() {
    println!("{}", factorial(8));
}

#[trace_var(p, n)]
fn factorial(mut n: u64) -> u64 {
    let mut p = 1;
    while n > 1 {
        p *= n;
        n -= 1;
    }
    p
}
```
output:
```rust
// Recursive expansion of trace_var macro
// =======================================

fn factorial(mut n: u64) -> u64 {
    let mut p = 1;
    while n > 1 {
        p *= n;
        n -= 1;
    }
    p
}
```

**Comment by lnicola** [comment]
It works for me, did you change `rust-analyzer.procMacro.ignored` and do you have any error in the logs?

![image](https://github.com/rust-lang/rust-analyzer/assets/308347/3d4263a9-e0a7-45ff-96d5-9e14bb7cdea7)

**Comment by qinchuan-dev** [comment]
which value do you set the config item?

**Comment by lnicola** [comment]
I didn't, I left it as default (`{}`).

**Comment by qinchuan-dev** [comment]
in windows, nvim and vscode both expand the macro rightly.
but in wsl, nvim and vscode don't. 
I will check my dev enviroment.

**Comment by lnicola** [comment]
Does running `cargo clean` and reopening the project help (just in case you're opening the same directory in both places)?

**Comment by qinchuan-dev** [comment]
for neovim, they don't share the same storage. they are deployed seperately and use the same vim plugs, same init.vim and same coc-settings.json

for vscode,  all config are defaults.

**Comment by qinchuan-dev** [comment]
> Does running `cargo clean` and reopening the project help (just in case you're opening the same directory in both places)?

THIS WORKS!

In WSL:
cargo clean -> nvim open -> rust expand macro  -> OK
cargo clean -> vscode open -> rust expand macro  -> OK

cargo clean -> vscode open -> rust expand macro -> OK -> vscode close -> vim open -> BAD

In Windows: 
All cases are OK.

**Comment by lnicola** [comment]
Yeah, sorry, I don't know why that happens. Maybe set `RA_LOG_FILE=foo` when running NVim (I'm sure you can access the logs some other way).
